src/corecel/sys/Environment.cc

```diff
@@ -34,11 +47,56 @@ Environment& environment()
  */
 std::string const& getenv(std::string const& key)
 {
-    static std::mutex getenv_mutex;
-    std::lock_guard<std::mutex> scoped_lock{getenv_mutex};
+    std::lock_guard<std::mutex> scoped_lock{getenv_mutex()};
```

Comment:
`std::scoped_lock` is the most recent implementation.

```suggestion
    std::scoped_lock<std::mutex> scoped_lock{getenv_mutex()};
```

Reply:
Done for this file, added to https://github.com/celeritas-project/celeritas/issues/1263 for the rest of the codebase.